### PR TITLE
Also match on release files without query strings

### PR DESCRIPTION
**Pull request by mitsuhiko**
This changes the logic for release file matching so that it will also
try to resolve files without query string.

This is necessary for react-native where all minified files add some
garbage into the query string for the minified file.

### CODE REVIEW

**Comment by mitsuhiko** [comment]
If we want to do this I will add some tests. The alternative is to force people to preprocess all js files to not include those query strings in the references.

File: src/sentry/lang/javascript/processor.py [hunk 1]
```diff
@@ -223,14 +223,23 @@ def fetch_release_file(filename, release):
         md5_text(filename).hexdigest(),
     )
 
-    filename_path = None
+    filename_idents = [ReleaseFile.get_ident(filename)]
+
     if filename is not None:
+        filename_no_query = filename.split('?')[0]
```
Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: Might as well do `filename.split('?', 1)[0]`

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
Need a CHANGES for this even though Danger didn't say so. This is a worthy change in behavior.

And a test would be nice.

To clarify, this means we'll have now potenitally 4 idents being looked up? Given the input: `http://example.com/?foo`, we'll have the following:

* http://example.com/a.js?foo
* http://example.com/a.js
* ~/a.js?foo
* ~/a.js

**Comment by mitsuhiko** [comment]
If we want this behavior I will add tests for it. I was a little bit worried we might have too many requests at this point. One thing I also just realized is that for some clients we appear to accept non urls but filenames so we currently might look up `foo.html` and `~foo.html`. (If I interpret #4718 correctly. Is this the case?)

File: src/sentry/lang/javascript/processor.py (hunk 1 above)
Comment (mitsuhiko) [inline src/sentry/lang/javascript/processor.py]: Indeed.

**Comment by mattrobenolt** [comment]
> One thing I also just realized is that for some clients we appear to accept non urls but filenames so we currently might look up foo.html and ~foo.html. (If I interpret #4718 correctly. Is this the case?)

Unsure off hand. I'd need to read this mess of code. 😞

**Comment by benvinegar** [comment]
Refs #3744

**Comment by benvinegar** [comment]
Hey folks – how do we revive this PR? Another user has hit this issue [on the forums](https://forum.sentry.io/t/sourcemaps-trouble-getting-them-working-consistently/923).

**Comment by benvinegar** [comment]
>  I was a little bit worried we might have too many requests at this point.

In cases where we drop the query string, I think these should be artifact lookups solely.

Or put another way: the only HTTP request we make is for the original, full URL we recovered for the frame. For artifacts lookups, we can expand the search space.

**Comment by mitsuhiko** [comment]
@benvinegar I'm happy if we merge this. The reason we did not end up doing it yet is because `sentry-cli` now writes out sourcemap references with the uploaded .min.js files which also resolves that issue.

**Comment by mitsuhiko** [comment]
If we end up including this we need to add tests.

**Comment by benvinegar** [comment]
Boop

**Comment by mattrobenolt** [comment]
@mitsuhiko I'm gonna take this over and carry it through. Swore this was merged.

**Comment by mattrobenolt** [comment]
Superseded by #5860

**Pull request closed without merging**